pkg/apis/argoproj/v1alpha1/argocd_types.go

```diff
@@ -354,6 +354,11 @@ const (
 type ArgoCDSSOSpec struct {
 	// Provider installs and configures the given SSO Provider with Argo CD.
 	Provider SSOProviderType `json:"provider,omitempty"`
+	// InsecureMode should be set to true in the below cases.
+	// 1. Keycloak service is not accesible by service name due to Network policies.
+	// 2. Operator is run locally for development purpose and default Ingress Controller which generates self-signed certificates
+	// is used as the CA authority.
+	InsecureMode bool `json:"insecureMode,omitempty"`
```

Comment:
`InsecureMode` is basically to turn off cert validation in the tls connecction between the keycloak api server, right?   I think we can just name the field `VerifyTLS` and default it to true.

Reply:
The reason behind naming it as InsecureMode is to provide an insight that it should mostly not be used. I believe customers usually have their clusters setup with certificates signed by a valid CA. So it should not be used even if the operator is run using the operator-sdk outside the cluster for development purpose. 

Please suggest.

Reply:
People know tlsverify should not be disabled in production.   We already use  `TLSVerify` some other struct in the CR.   `InsecureMode` and the comment there do not tell me what it does.   Someone will have to look into the code to figure out.   In the end, it is just a TLSVerify flag plus time sent on looking at the code to see what it does.

Reply:
ack ... made changes accordingly :)